src/elements/ngm-share-link.js

```diff
@@ -0,0 +1,110 @@
+import {LitElementI18n} from '../i18n';
+import {html} from 'lit-element';
+import {showWarning} from '../message';
+import i18next from 'i18next';
+import $ from '../jquery';
+import 'fomantic-ui-css/components/popup.js';
+import './ngm-i18n-content.js';
+import {SHORTLINK_HOST_BY_PAGE_HOST} from '../constants';
+
+class NgmShareLink extends LitElementI18n {
+
+  static get properties() {
+    return {
+      shortlink: {type: String}
+    };
+  }
+
+  constructor() {
+    super();
+
+    this.shortlink = window.location.href;
+    this.getShortlink().then(link => {
+      this.shortlink = link;
+    });
+  }
+
+  firstUpdated() {
+    const shortlinkPopupKey = navigator.clipboard ? 'welcome_shortlink_copied' : 'welcome_copy_guide';
+    // workaround: t('welcome_shortlink_copied')
+    // workaround: t('welcome_copy_guide')
+    $(this.querySelector('.ngm-share-link > input')).popup({
+      on: 'manual',
+      position: 'bottom center',
+      html: `<ngm-i18n-content key="${shortlinkPopupKey}"></ngm-i18n-content>`,
+      variation: 'mini',
+      hideOnScroll: true
+    });
+  }
+
+  async getShortlink() {
+    const serviceHost = SHORTLINK_HOST_BY_PAGE_HOST[window.location.host];
+    try {
+      const result = await fetch(`https://${serviceHost}/admin_shrink_url`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json;charset=UTF-8'
+        },
+        body: JSON.stringify({
+          'url_long': window.location.href,
```

Comment:
It looks like the admin_shrink_url service accepts any URL. Would you be able to accept only our domains.

Reply:
Do you mean an update of the lambda function?

Reply:
Yes

Reply:
I think we should create an issue for this because it will take some time. Simon propose to make some security key for requests but maybe make host restriction will be more easily